mantidproject/mantid
# Fix color map editor in vsi

This fixes issue [#11717](http://trac.mantidproject.org/mantid/ticket/11717).

**Context**

Following the migration to PV4.3.1 the color editor panel has been changed. It requires a dockable window. Users noticed that it was on top of the actual rendered view. It was mentioned that it should be a separate window.

In addition it was noted that the new feature of the opacity function mapper was not updating correctly when a user changed a range in custom line edit fields for min and max of the scalar range.
### For testing

**Test that dock window is not docking**
1. Open a [sample](http://trac.mantidproject.org/mantid/attachment/ticket/11717/MDEvent_Osiris.nxs) workpspace into the VSI
2. Click on the color editor panel button (The edit button in the properties panel)
-  Confirm that the panel opens in a separate window and that it is positioned in the left top corner of the VSI
  1. Move the window, close it and reopen it.
-  Confirm that the panel appears at the same location again

Note that when double clicking the top panel of the editor window, the window will hide. This is not a bug, but a workaround.

**Test the opacity function**
1. Open the sample workspace into the VSI
2. Open the color editor panel. You see two color widgets, one for the opacity and one for the color mapping itself. 
- Confirm that that both widgets show the specified range (you need to click on the ends to see this)
  1. Uncheck auto update of the color scale on the VSI and manually set the scale range to something lower than currently set
- Confirm that the opacity function editor udpates
  1. Switch the view
- Confirm that the opacity function editor does not change
